Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -109,7 +121,18 @@ function Get-DiagnosticsExtensions($storageAccount, $extensionsPath, $storageAcc
                         {
                             try
                             {
-                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName
+                                $publicConfigStorageKey = Get-AzureStorageKey -StorageAccountName $publicConfigStorageAccountName -ErrorAction Ignore
+                                if ($publicConfigStorageKey -eq $null)
+                                {
+                                    #if the storage account is not the classic one, get the primary key in another way
+                                    $publicConfigStorageKey = (Get-AzureRmStorageAccountKey -ResourceGroupName "$storageAccountResourceGroup" -Name "$publicConfigStorageAccountName").Value[0]
```

Comment:
1. Even if you logged in with connection type classic still your cmdlet for ARM worked because for connection type ARM you already being loggedin as you previously might have run some ARM cmdlets with ARM connection type.

2. For tasking ARM input in "AzureCloudPowerShellDeployment" task you can refer code of AzurePS task or AzureSqlDacpac task. But it will not solve problem as we are discussing but to ensure you can give a try.
Refer: https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/SqlAzureDacpacDeployment/task.json#L43-L72
https://github.com/Microsoft/vsts-tasks/blob/master/Tasks/SqlAzureDacpacDeployment/DeploySqlAzure.ps1#L112-L115